03_oop_part_2/src/item.ts

```diff
@@ -0,0 +1,33 @@
+import { Page } from './page';
+import { Pages } from './pages';
+
+
+const PagesIterable = () => class implements IterableIterator<Item> {
+    constructor(protected pages: Pages, protected openPage?: Page) {}
+
+    public next(): IteratorResult<Item> {
+        let nxt = this.pages.next();
+        if (!nxt.done) {
+            this.openPage = nxt.value;
+            return {done: false, value: this};
+        }
+        this.openPage = undefined;
+        return {done: true, value: null};
+    }
+
+    [Symbol.iterator](): IterableIterator<Item> {
+        return this;
+    }
+
+    protected get pageSuffix(): String {
```

Comment:
Looks a bit dirty, not an OOP way :slightly_smiling_face:.
Hint: PagesIterable should implement `IterableIterator<string>`, not `IterableIterator<Item>`.